src/box/box.cc

```diff
@@ -5969,14 +5970,24 @@ box_init(void)
 	crash_callback = box_crash_callback;
 }
 
+#ifdef TEST_BUILD
+double box_client_shutdown_timeout = TIMEOUT_INFINITY;
+#else
+double box_client_shutdown_timeout = 3.0;
```

Comment:
Nit:
- I wouldn't define the variable twice - this may be confusing.
- A comment is missing. Would be nice to have an explanation why it's infinity on test builds.
- Better name it simply `box_shutdown_timeout`.
- Should be static and better move it to the top, where other constants are defined.

```C
#ifdef TEST_BUILD
# define BOX_SHUTDOWN_TIMEOUT_DEFAULT TIMEOUT_INFINITY
#else
# define BOX_SHUTDOWN_TIMEOUT_DEFAULT 3.0
#endif

/** TODO: comment */
static double box_shutdown_timeout = BOX_SHUTDOWN_TIMEOUT_DEFAULT;
```

Reply:
Done.

<details>
<summary>diff</summary>

```diff
diff --git a/src/box/box.cc b/src/box/box.cc
index d2710f86b..1225d25d8 100644
--- a/src/box/box.cc
+++ b/src/box/box.cc
@@ -247,6 +247,23 @@ static char box_feedback_host[BOX_FEEDBACK_HOST_MAX];
 /** Whether sending crash info to feedback URL is enabled. */
 static bool box_feedback_crash_enabled;
 
+#ifdef TEST_BUILD
+/**
+ * Set timeout to infinity in test build because first not all CI tests treat
+ * non zero exit code of Tarantool instance as failure currently. Also in
+ * luatest currently it is easier to test for no hanging rather then for
+ * Tarantool instance exit code.
+ */
+#define BOX_SHUTDOWN_TIMEOUT_DEFAULT TIMEOUT_INFINITY
+#else
+#define BOX_SHUTDOWN_TIMEOUT_DEFAULT 3.0
+#endif
+
+/** Timeout on waiting client related fibers to finish. */
+static double box_shutdown_timeout = BOX_SHUTDOWN_TIMEOUT_DEFAULT;
+TWEAK_DOUBLE(box_shutdown_timeout);
+
+
 static int
 box_run_on_recovery_state(enum box_recovery_state state)
 {
@@ -5970,13 +5987,6 @@ box_init(void)
 	crash_callback = box_crash_callback;
 }
 
-#ifdef TEST_BUILD
-double box_client_shutdown_timeout = TIMEOUT_INFINITY;
-#else
-double box_client_shutdown_timeout = 3.0;
-#endif
-TWEAK_DOUBLE(box_client_shutdown_timeout);
-
 /** Shutdown box storage i.e. stop parts that need TX loop running. */
 static void
 box_storage_shutdown()
@@ -5984,7 +5994,7 @@ box_storage_shutdown()
 	if (!is_storage_initialized)
 		return;
 	is_storage_shutdown = true;
-	if (iproto_shutdown(box_client_shutdown_timeout) != 0) {
+	if (iproto_shutdown(box_shutdown_timeout) != 0) {
 		diag_log();
 		panic("cannot gracefully shutdown iproto");
 	}
diff --git a/test/box-luatest/shutdown_test.lua b/test/box-luatest/shutdown_test.lua
index b4fc5c7be..b59cb3791 100644
--- a/test/box-luatest/shutdown_test.lua
+++ b/test/box-luatest/shutdown_test.lua
@@ -104,7 +104,7 @@ g.test_shutdown_of_hanging_iproto_request = function(cg)
             local log = require('log')
             local fiber = require('fiber')
             local tweaks = require('internal.tweaks')
-            tweaks.box_client_shutdown_timeout = 1.0
+            tweaks.box_shutdown_timeout = 1.0
             log.info('going to sleep for test')
             while true do
                 pcall(fiber.sleep, 1000)
```

</details>